graalvm-native-image-app/build.gradle

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id 'application'
+    id 'org.graalvm.buildtools.native' version '0.9.23'
+}
+
+application {
+    mainClass = 'com.mongodb.internal.graalvm.NativeImageApp'
+}
+
+// see https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html
+graalvmNative {
+    metadataRepository {
```

Comment:
`metadataRepository` is enabled by default, and I configured it explicitly only to make sure we depend on a fixed version, to make our builds reproducible.

However, because of you question, I disabled it, and everything still works, which means that when we collect the reachability metadata, we happen to collect everything that is then needed at runtime, so the data from the metadata repository never comes into play. The latter is actually obvious, since we execute exactly the same code when we are collecting the metadata and when we test the native app.